client/a8c-for-agencies/lib/permission.ts

```diff
@@ -98,14 +98,6 @@ export const isPathAllowed = ( pathname: string, agency: Agency | null ) => {
 		return false;
 	}
 
-	// Admins can access all paths
-	const role = agency?.user?.role;
-	// For some cases, the role isn't set, so we default to 'a4a_administrator'
-	// FIXME: This should be removed after the role is set for all users
-	if ( ! role || role === 'a4a_administrator' ) {
```

Comment:
I think it's better to keep the `role === 'a4a_administrator'` check as the admin can see all the pages, and we need to do an unnecessary check. What do you think?

Reply:
@yashwin We need to remove that check as well because of the agency level restrictions we're implementing here: https://github.com/Automattic/wp-calypso/pull/94999. 

Depending on the agency's tier, an admin might not have access to all pages so we do need do the check.